### PR TITLE
Add an AssertLexemes method to simplify LexerTests

### PR DESCRIPTION
**Pull request by ahoppen**

### CODE REVIEW

**Comment by ahoppen** [comment]
@swift-ci Please test

**Comment by ahoppen** [comment]
@swift-ci Please test

File: Tests/SwiftParserTest/Assertions.swift [hunk 1]
```diff
@@ -29,45 +59,85 @@ import _SwiftSyntaxTestSupport
...
+  }
+  var source = source
+  source.withUTF8 { buf in
+    var lexemes = [Lexer.Lexeme]()
+    for token in Lexer.tokenize(buf, from: 0) {
+      lexemes.append(token)
+
+      guard token.tokenKind != .eof else {
+        break
+      }
```
Comment (bnbarham) [inline Tests/SwiftParserTest/Assertions.swift:138]: I'd find:
```
if token.tokenKind == .eof {
  break
}
```
much easier to parse, maybe that's just me though 😅

**Review by bnbarham — APPROVED** [review]

File: Tests/SwiftParserTest/Assertions.swift (hunk 1 above)
Comment (ahoppen) [reply Tests/SwiftParserTest/Assertions.swift:138]: I agree but I just copied this from the old `Lexer.lex` method in LexerTests.swift.

I’ll update it in a follow-up PR.

**Pull request merged**